### PR TITLE
Move how to setup and run PuppetDB unit tests

### PR DESCRIPTION
**Pull request by briancain**
This pull request moves how to set up postgresql to run the unit tests
into the CONTRIBUTING doc. It also includes the contributing document
in the navigation header.

### CODE REVIEW

File: documentation/install_from_source.markdown [hunk 1]
```diff
@@ -8,6 +8,7 @@ layout: default
 [configure_heap]: ./configure.html#configuring-the-java-heap-size
 [module]: ./install_via_module.html
 [packages]: ./install_from_packages.html
+[running_tests]: ./CONTRIBUTING.html#running-the-tests
```
Comment (briancain) [inline documentation/install_from_source.markdown]: I'm not entire sure if this is right.....I'm guessing the tool that deploys these docs just takes the markdown files and translates them to .html files?

**Comment by ajroetker** [comment]
This lgtm

File: documentation/CONTRIBUTING.md [hunk 2]
```diff
@@ -41,11 +48,41 @@ top of things.
...
+Then you can run the test suite:
+
+    $ lein test
+
+And if you'd like to preserve the temporary test databases on failure, you can
+set `PDB_TEST_PRESERVE_DB_ON_FAIL` to true:
+
+```
+$ PDB_TEST_KEEP_DB_ON_FAIL=true lein test
+```
```
Comment (wkalt) [inline documentation/CONTRIBUTING.md]: does this work on the site?

File: documentation/install_from_source.markdown (hunk 1 above)
Comment (wkalt) [inline documentation/install_from_source.markdown]: yeah, this should be fine.

File: documentation/CONTRIBUTING.md (hunk 2 above)
Comment (ajroetker) [inline documentation/CONTRIBUTING.md]: What wouldn't work?

**Pull request merged**